### DISCUSSION TITLE
Remote XPC (Tunnel) Q&A

### DISCUSSION THREAD

**Discussion by morellexf13** [doronz88/pymobiledevice3#565] [General]
Hi @doronz88, 

- Is there any way to close or kill a tunnel? For example: I have the following --rsd fd12:b60:e82d::1 57120, how can I programatically end this connection?

- Is there any limit on how many tunnels are created?

**Comment by doronz88** [comment]
- In general, the CLI acts as a bunch of biolerplate code snippets to do anything. In your case, you can just copy the same usage as demonstrated in the `start-quic-tunnel` implementation: https://github.com/doronz88/pymobiledevice3/blob/master/pymobiledevice3/cli/remote.py#L100
- Not that I know of, except for the number of allowed tunnels allowed on the OS itself. You can reuse the same tunnel for many different connections - it's basically a VPN instance.
